opm/parser/eclipse/EclipseState/Schedule/Well.cpp

```diff
@@ -303,11 +526,109 @@ namespace Opm {
         }
         return wellNameInPattern;
     }
+    bool Well::isMultiSegment() const {
+        return m_is_multi_segment;
+    }
+
+    void Well::setMultiSegment(const bool is_multi_segment) {
+        m_is_multi_segment = is_multi_segment;
+    }
+
+    SegmentSetConstPtr Well::getSegmentSet(size_t time_step) const {
+        return m_segmentset->get(time_step);
+    }
+
 
     bool Well::getAllowCrossFlow() const {
         return m_allowCrossFlow;
     }
 
+    void Well::processCOMPSEGS(size_t time_step, std::vector<CompsegsPtr>& compsegs) {
```

Comment:
1. `m_segmentset` contains the segment structure and some information about each segment. 
2. `std::vector<CompsegsPtr>& compsegs` contains the location information of each completion in the multi-segment wells. 
3. The function firstly based on the location information from `std::vector<CompsegsPtr>& compsegs` find the segment that the completion should belong to and update the center-depth based on the segment information. (in multi-segment wells, the depth of the perforation is not that related to the cell center anymore). 
4. With updated information from `std::vector<CompsegsPtr>& compsegs`, update `m_completions`.